addons/account/models/account_move.py

```diff
@@ -1320,24 +1320,63 @@ def _compute_payments_widget_to_reconcile_info(self):
 
-            domain = [
-                ('account_id', 'in', pay_term_lines.account_id.ids),
-                ('parent_state', '=', 'posted'),
-                ('partner_id', '=', move.commercial_partner_id.id),
-                ('reconciled', '=', False),
-                '|', ('amount_residual', '!=', 0.0), ('amount_residual_currency', '!=', 0.0),
-            ]
+            def _get_common_domain():
+                """ Return the common domain conditions for searching outstanding payments """
+                return [
+                    ('parent_state', '=', 'posted'),
+                    ('partner_id', '=', move.commercial_partner_id.id),
+                    ('reconciled', '=', False),
+                    '|', ('amount_residual', '!=', 0.0), ('amount_residual_currency', '!=', 0.0),
+                ]
+
+            # Construct the base domain
+            domain = expression.AND([_get_common_domain(), [('account_id', 'in', pay_term_lines.account_id.ids)]])
+
+            # Bank line domain
+            bank_journal_ids = self.env['account.journal'].search_fetch(
+                [
+                    *self.env['account.journal']._check_company_domain(move.company_id),
+                    ('type', '=', 'bank'),
+                ],
+                ['id'],
+            ).ids
+
+            bank_domain = expression.AND([_get_common_domain(), [
+                ('account_id.account_type', 'in', ['asset_cash', 'asset_receivable']),
+                ('journal_id', 'in', bank_journal_ids),
+                ('partner_id', '!=', False),
+                ('balance', '>', 0.0)
+            ]])
 
             payments_widget_vals = {'outstanding': True, 'content': [], 'move_id': move.id}
 
             if move.is_inbound():
                 domain.append(('balance', '<', 0.0))
+                domain = expression.OR([domain, bank_domain])
                 payments_widget_vals['title'] = _('Outstanding credits')
             else:
                 domain.append(('balance', '>', 0.0))
                 payments_widget_vals['title'] = _('Outstanding debits')
 
-            for line in self.env['account.move.line'].search(domain):
+            lines = self.env['account.move.line'].search(domain)
+
+            move_names = lines.mapped('move_name')
+
+            reconciled_counts = {
+                rec[0]: rec[1]
+                for rec in self.env['account.move.line']._read_group(
+                    [('move_name', 'in', move_names), ('reconciled', '=', True)],
+                    ['move_name'],
+                    ['__count']
+                )
+            }
+
+            for line in lines:
+                if line.account_id.account_type == 'asset_cash':
+                    if any(
+                        l for l in lines if l != line and l.move_name == line.move_name
+                    ) or reconciled_counts.get(line.move_name, 0) > 0:
```

Comment:
This any feels a bit weird to me, specially that we have to go through lines all over again, wondering if we can do better here :thinking: 
also i think that reconciled_counts.get(line.move_name) is enough ? No need for the > 0 since the get will return false if he don't find the element, and if it finds it it will a number > 0 and will be interpret as a true element by python